neqo-transport/src/send_stream.rs

```diff
@@ -563,20 +564,26 @@ impl SendStream {
             .map_or(0, |tx| self.max_stream_data - tx.data_limit())
     }
 
-    /// Bytes sendable on stream. Constrained by both stream credit available
-    /// and space in the tx buffer.
+    /// Bytes sendable on stream. Constrained by stream credit available,
+    /// connection credit available, and space in the tx buffer.
     pub fn avail(&self) -> u64 {
-        self.state
+        let stream_avail = self
+            .state
             .tx_buf()
-            .map_or(0, |tx| min(self.credit_avail(), tx.avail() as u64))
+            .map_or(0, |tx| min(self.credit_avail(), tx.avail() as u64));
+        min(stream_avail, self.flow_mgr.borrow().conn_credit_avail())
```

Comment:
Is this simpler as

```rust
min(min(self.state.tx_buf().unwrap_or(0) as u64, self.credit_avail()), self.flow_mgr.borrow().conn_credit_avail())
```

Maybe with some extra lines added for readability.  And something to suppress the warnings from the cast.

Reply:
It's not verbatim what you suggested but I took the internal min() out from inside the map_or() and it seems more readable to me.

there's also:
```
    *[
        self.state.tx_buf().map_or(0, |tx| tx.avail() as u64),
        self.credit_avail(),
        self.flow_mgr.borrow().conn_credit_avail(),
    ]
    .iter()
    .min()
    .unwrap()
```
but that feels a bit overkill for just three values.